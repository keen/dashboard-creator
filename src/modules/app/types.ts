export type ViewMode = 'editor' | 'viewer' | 'management';

export type ReducerState = {
  view: ViewMode;
  activeDashboardId: string | null;
  cachedDashboardsNumber: number;
  user: {
    editPrivileges: boolean;
  };
  imagePicker: {
    isVisible: false;
  };
  visualizationEditor: {
    isVisible: false;
  };
  queryPicker: {
    isVisible: boolean;
  };
};
