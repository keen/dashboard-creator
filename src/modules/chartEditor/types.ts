import { Query } from '@keen.io/query';
import {
  PickerWidgets,
  ChartSettings,
  WidgetSettings,
} from '@keen.io/widget-picker';
import { TextSettings } from '@keen.io/widgets';

export enum EditorSection {
  QUERY = 'query',
  SETTINGS = 'settings',
}

export type EditorWidgetSettings = WidgetSettings & {
  title: TextSettings;
  subtitle: TextSettings;
};

export type ReducerState = {
  editorSection: EditorSection;
  isOpen: boolean;
  isEditMode: boolean;
  isSavedQuery: boolean;
  isDirtyQuery: boolean;
  isQueryPerforming: boolean;
  hasQueryChanged: boolean;
  changeQueryConfirmation: boolean;
  querySettings: Partial<Query>;
  initialQuerySettings: Partial<Query>;
  analysisResult: Record<string, any> | null;
  queryError: string | null;
  visualization: {
    type: Exclude<PickerWidgets, 'json'>;
    chartSettings: ChartSettings;
    widgetSettings: EditorWidgetSettings;
  };
};

export type VisualisationSettingsPayload = {
  type: PickerWidgets;
  chartSettings: ChartSettings;
  widgetSettings: Record<string, any>;
};
