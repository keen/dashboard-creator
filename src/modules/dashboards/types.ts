import { Theme } from '@keen.io/charts';
import { Widget } from '../widgets';

export type DashboardMetaData = {
  id: string;
  title: null | string;
  widgets: number;
  queries: number;
  tags: string[];
  lastModificationDate: number;
  isPublic: boolean;
  publicAccessKey: null | string;
};

export type DashboardSettings = {
  colorPalette: string;
  page: {
    gridGap: number;
    background: string;
    chartTitlesFont: string;
    visualizationsFont: string;
  };
  tiles: {
    background: string;
    borderColor: string;
    borderRadius: number;
    borderWidth: number;
    padding: number;
    hasShadow: boolean;
  };
};

export type DashboardModel = {
  version: string;
  widgets: Widget[];
  settings?: DashboardSettings;
  theme?: Partial<Theme>;
};

export type Dashboard = {
  version: string;
  widgets: string[];
};

export enum DashboardError {
  NOT_EXIST = 'NOT_EXIST',
  ACCESS_NOT_PUBLIC = 'ACCESS_NOT_PUBLIC',
  VIEW_PUBLIC_DASHBOARD = 'VIEW_PUBLIC_DASHBOARD',
}

export type DashboardItem = {
  initialized: boolean;
  isSaving: boolean;
  settings: Dashboard;
  error: null | DashboardError;
};

export type DashboardListOrder = 'recent' | 'oldest' | 'az' | 'za';

export type ReducerState = {
  deleteConfirmation: {
    isVisible: boolean;
    dashboardId?: string;
  };
  metadata: {
    isInitiallyLoaded: boolean;
    error: boolean;
    data: DashboardMetaData[];
    isSavingMetadata: boolean;
    isRegeneratingAccessKey: boolean;
  };
  dashboardSettingsModal: {
    isVisible: boolean;
    dashboardId: string | null;
  };
  dashboardShareModal: {
    isVisible: boolean;
    dashboardId: string | null;
  };
  tagsPool: string[];
  tagsFilters: {
    showOnlyPublicDashboards: boolean;
    tags: string[];
  };
  items: Record<string, DashboardItem>;
  dashboardListOrder: DashboardListOrder;
  cachedDashboardIds: string[];
};
