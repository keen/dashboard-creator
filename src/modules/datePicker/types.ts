export type ReducerState = {
  isEditorOpen: boolean;
  widgetConnections: DatePickerConnection[];
};

export type DatePickerConnection = {
  widgetId: string;
  isConnected: boolean;
  title: string | null;
  positionIndex: number;
};
