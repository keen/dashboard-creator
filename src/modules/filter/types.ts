export type ReducerState = {
  isEditorOpen: boolean;
  widgetConnections: FilterConnection[];
  eventStream: string;
  targetProperty: string;
};

export type FilterConnection = {
  widgetId: string;
  isConnected: boolean;
  title: string | null;
  positionIndex: number;
};
