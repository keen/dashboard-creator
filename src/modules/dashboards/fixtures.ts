import { DashboardMetaData } from './types';

export const dashbordsMeta: DashboardMetaData[] = [
  {
    id: '@dashboard/01',
    widgets: 5,
    queries: 0,
    title: 'Dashboard 1',
    isPublic: true,
  },
  {
    id: '@dashboard/02',
    widgets: 0,
    queries: 2,
    title: 'Dashboard 2',
    isPublic: true,
  },
];
