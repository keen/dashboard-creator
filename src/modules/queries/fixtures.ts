/* eslint-disable @typescript-eslint/naming-convention */

import { SavedQuery, SavedQueryAPIResponse } from './types';

export const savedQueriesResponse: SavedQueryAPIResponse[] = [
  {
    query: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'logins',
      order_by: null,
    },
    refresh_rate: 0,
    query_name: '@query/01',
    metadata: {
      display_name: 'Query 01',
      visualization: {
        type: 'bar',
        chart_settings: {},
        widget_settings: {},
      },
    },
  },
  {
    query: {
      analysis_type: 'sum',
      timeframe: 'this_14_days',
      event_collection: 'purchases',
      order_by: null,
    },
    refresh_rate: 0,
    query_name: '@query/02',
    metadata: {
      display_name: 'Query 02',
      visualization: {
        type: 'area',
        chart_settings: {},
        widget_settings: {},
      },
    },
  },
];

export const savedQueries: SavedQuery[] = [
  {
    id: '@query/01',
    displayName: 'Query 01',
    visualization: {
      type: 'bar',
      chartSettings: {},
      widgetSettings: {},
    },
    settings: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'purchases',
      order_by: null,
    },
  },
  {
    id: '@query/02',
    displayName: 'Query 02',
    visualization: {
      type: 'metric',
      chartSettings: {},
      widgetSettings: {},
    },
    settings: {
      analysis_type: 'count',
      timeframe: 'this_14_days',
      event_collection: 'pageviews',
      order_by: null,
    },
  },
];
