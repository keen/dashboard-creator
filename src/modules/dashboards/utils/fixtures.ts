import { DashboardMetaData } from '../../../modules/dashboards';

export const dashboardsMeta: DashboardMetaData[] = [
  {
    id: '@dashboard/01',
    widgets: 5,
    queries: 0,
    title: 'Dashboard 1',
    tags: ['tag-1', 'tag-2', 'tag-4'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/1',
  },
  {
    id: '@dashboard/02',
    widgets: 0,
    queries: 2,
    title: 'Dashboard 2',
    tags: ['tag-2', 'tag-3'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/2',
  },
  {
    id: '@dashboard/03',
    widgets: 0,
    queries: 0,
    title: null,
    tags: ['tag-1', 'tag-4'],
    lastModificationDate: 1606895352390,
    isPublic: true,
    publicAccessKey: '@public/3',
  },
];
