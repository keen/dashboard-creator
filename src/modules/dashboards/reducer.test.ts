import { dashboardsActions, dashboardsReducer, initialState } from './index';
import { dashboardsMeta } from './fixtures';

import { DashboardError } from './types';

test('creates dashboard', () => {
  const action = dashboardsActions.createDashboard('@dashboard/01');
  const {
    metadata: { data },
  } = dashboardsReducer(initialState, action);

  expect(data).toMatchObject([
    {
      id: '@dashboard/01',
      isPublic: false,
      publicAccessKey: null,
      queries: 0,
      tags: [],
      title: null,
      widgets: 0,
    },
  ]);
});

test('updates cached dashboards identifiers', () => {
  const action = dashboardsActions.updateCachedDashboardIds(['@dashboard/01']);
  const { cachedDashboardIds } = dashboardsReducer(initialState, action);

  expect(cachedDashboardIds).toMatchInlineSnapshot(`
    Array [
      "@dashboard/01",
    ]
  `);
});

test('set dashboard error', () => {
  const state = {
    ...initialState,
    items: {
      '@dashboard/01': {
        initialized: false,
        isSaving: false,
        error: null,
        settings: null,
      },
    },
  };

  const action = dashboardsActions.setDashboardError({
    dashboardId: '@dashboard/01',
    error: DashboardError.ACCESS_NOT_PUBLIC,
  });
  const { items } = dashboardsReducer(state, action);

  expect(items).toMatchInlineSnapshot(`
    Object {
      "@dashboard/01": Object {
        "error": "ACCESS_NOT_PUBLIC",
        "initialized": false,
        "isSaving": false,
        "settings": null,
      },
    }
  `);
});

test('updates dashboard save indicator', () => {
  const state = {
    ...initialState,
    items: {
      '@dashboard/01': {
        initialized: false,
        isSaving: false,
        error: null,
        settings: null,
      },
    },
  };

  const action = dashboardsActions.saveDashboard('@dashboard/01');
  const { items } = dashboardsReducer(state, action);

  expect(items).toMatchInlineSnapshot(`
    Object {
      "@dashboard/01": Object {
        "error": null,
        "initialized": false,
        "isSaving": true,
        "settings": null,
      },
    }
  `);
});

test('updates dashboard metadata and sort it by recent', () => {
  const action = dashboardsActions.updateDashboardMetadata({
    dashboardId: '@dashboard/01',
    metadata: {
      title: 'Title',
      queries: 30,
    },
  });
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      isLoaded: true,
      data: [
        {
          id: '@dashboard/01',
          widgets: 1,
          queries: 0,
          title: null,
          lastModificationDate: 0,
          tags: [],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
        {
          id: '@dashboard/02',
          widgets: 1,
          queries: 0,
          title: null,
          lastModificationDate: 1,
          tags: [],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
      ],
    },
  };

  const { metadata } = dashboardsReducer(state, action);

  expect(metadata).toMatchInlineSnapshot(`
    Object {
      "data": Array [
        Object {
          "id": "@dashboard/02",
          "isPublic": false,
          "lastModificationDate": 1,
          "publicAccessKey": null,
          "queries": 0,
          "savedQueries": Array [],
          "tags": Array [],
          "title": null,
          "widgets": 1,
        },
        Object {
          "id": "@dashboard/01",
          "isPublic": false,
          "lastModificationDate": 0,
          "publicAccessKey": null,
          "queries": 30,
          "savedQueries": Array [],
          "tags": Array [],
          "title": "Title",
          "widgets": 1,
        },
      ],
      "error": null,
      "isInitiallyLoaded": false,
      "isLoaded": true,
      "isRegeneratingAccessKey": false,
      "isSavingMetadata": false,
    }
  `);
});

test('register a new dashboard', () => {
  const action = dashboardsActions.registerDashboard('@dashboard/01');
  const { items } = dashboardsReducer(initialState, action);

  expect(items).toMatchInlineSnapshot(`
    Object {
      "@dashboard/01": Object {
        "error": null,
        "initialized": false,
        "isSaving": false,
        "settings": null,
      },
    }
  `);
});

test('updates settings for dashboard', () => {
  const state = {
    ...initialState,
    items: {
      '@dashboard/01': {
        initialized: false,
        isSaving: false,
        error: null,
        settings: null,
      },
    },
  };
  const dashboardSettings = {
    version: '0.0.1',
    widgets: ['@widget/01'],
  };

  const action = dashboardsActions.updateDashboard({
    dashboardId: '@dashboard/01',
    settings: dashboardSettings,
  });
  const { items } = dashboardsReducer(state, action);

  expect(items).toMatchInlineSnapshot(`
    Object {
      "@dashboard/01": Object {
        "error": null,
        "initialized": true,
        "isSaving": false,
        "settings": Object {
          "version": "0.0.1",
          "widgets": Array [
            "@widget/01",
          ],
        },
      },
    }
  `);
});

test('add widget to dashboard', () => {
  const action = dashboardsActions.addWidgetToDashboard({
    dashboardId: '@dashboard/01',
    widgetId: '@widget/01',
  });
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      isLoaded: true,
      data: [
        {
          id: '@dashboard/01',
          widgets: 0,
          queries: 0,
          title: null,
          lastModificationDate: 0,
          tags: [],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
      ],
    },
    items: {
      '@dashboard/01': {
        initialized: true,
        isSaving: false,
        error: null,
        settings: {
          version: '0.0.1',
          widgets: [],
        },
      },
    },
  };

  const updatedState = dashboardsReducer(state, action);

  expect(updatedState).toMatchSnapshot();
});

test('removes widget from dashboard', () => {
  const action = dashboardsActions.removeWidgetFromDashboard({
    dashboardId: '@dashboard/01',
    widgetId: '@widget/01',
  });
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      isLoaded: true,
      data: [
        {
          id: '@dashboard/01',
          widgets: 1,
          queries: 0,
          title: null,
          lastModificationDate: 0,
          tags: [],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
      ],
    },
    items: {
      '@dashboard/01': {
        initialized: true,
        isSaving: false,
        error: null,
        settings: {
          version: '0.0.1',
          widgets: ['@widget/01'],
        },
      },
    },
  };

  const updatedState = dashboardsReducer(state, action);

  expect(updatedState).toMatchSnapshot();
});

test('serializes dashboards metadata', () => {
  const action = dashboardsActions.fetchDashboardsListSuccess(dashboardsMeta);
  const { metadata } = dashboardsReducer(initialState, action);

  expect(metadata).toMatchInlineSnapshot(`
    Object {
      "data": Array [
        Object {
          "id": "@dashboard/01",
          "isPublic": true,
          "lastModificationDate": 1606895352390,
          "publicAccessKey": "@public/1",
          "queries": 0,
          "tags": Array [],
          "title": "Dashboard 1",
          "widgets": 5,
        },
        Object {
          "id": "@dashboard/02",
          "isPublic": true,
          "lastModificationDate": 1606895352390,
          "publicAccessKey": "@public/2",
          "queries": 2,
          "tags": Array [],
          "title": "Dashboard 2",
          "widgets": 0,
        },
        Object {
          "id": "@dashboard/03",
          "isPublic": true,
          "lastModificationDate": 1606895352390,
          "publicAccessKey": "@public/3",
          "queries": 0,
          "tags": Array [],
          "title": null,
          "widgets": 0,
        },
      ],
      "error": null,
      "isInitiallyLoaded": true,
      "isRegeneratingAccessKey": false,
      "isSavingMetadata": false,
    }
  `);
});

test('opens dashboard settings', () => {
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      data: dashboardsMeta,
    },
  };
  const action = dashboardsActions.showDashboardSettingsModal('@dashboard/01');
  const { dashboardSettingsModal } = dashboardsReducer(state, action);

  expect(dashboardSettingsModal).toMatchInlineSnapshot(`
    Object {
      "dashboardId": "@dashboard/01",
      "isVisible": true,
    }
  `);
});

test('closes dashboard settings', () => {
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      data: dashboardsMeta,
    },
  };
  const action = dashboardsActions.hideDashboardSettingsModal();
  const { dashboardSettingsModal } = dashboardsReducer(state, action);

  expect(dashboardSettingsModal).toMatchInlineSnapshot(`
    Object {
      "dashboardId": null,
      "isVisible": false,
    }
  `);
});

test('opens dashboard share modal', () => {
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      data: dashboardsMeta,
    },
  };
  const action = dashboardsActions.showDashboardShareModal('@dashboard/01');
  const { dashboardShareModal } = dashboardsReducer(state, action);

  expect(dashboardShareModal).toMatchInlineSnapshot(`
    Object {
      "dashboardId": "@dashboard/01",
      "isVisible": true,
    }
  `);
});

test('closes dashboard share modal', () => {
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      data: dashboardsMeta,
    },
  };
  const action = dashboardsActions.hideDashboardShareModal();
  const { dashboardShareModal } = dashboardsReducer(state, action);

  expect(dashboardShareModal).toMatchInlineSnapshot(`
    Object {
      "dashboardId": null,
      "isVisible": false,
    }
  `);
});

test('return state for saving dashboard metadata', () => {
  const action = dashboardsActions.saveDashboardMetadata({
    dashboardId: '@dashboardId',
    metadata: dashboardsMeta[0],
  });
  const { metadata } = dashboardsReducer(initialState, action);
  expect(metadata).toMatchInlineSnapshot(`
    Object {
      "data": Array [],
      "error": null,
      "isInitiallyLoaded": false,
      "isRegeneratingAccessKey": false,
      "isSavingMetadata": true,
    }
  `);

  const actionSuccess = dashboardsActions.saveDashboardMetadataSuccess();
  const { metadata: metadataSuccess } = dashboardsReducer(
    initialState,
    actionSuccess
  );
  expect(metadataSuccess).toMatchInlineSnapshot(`
    Object {
      "data": Array [],
      "error": null,
      "isInitiallyLoaded": false,
      "isRegeneratingAccessKey": false,
      "isSavingMetadata": false,
    }
  `);

  const actionError = dashboardsActions.saveDashboardMetadataError();
  const { metadata: metadataError } = dashboardsReducer(
    initialState,
    actionError
  );
  expect(metadataError).toMatchInlineSnapshot(`
    Object {
      "data": Array [],
      "error": null,
      "isInitiallyLoaded": false,
      "isRegeneratingAccessKey": false,
      "isSavingMetadata": false,
    }
  `);
});

test('set order for dashboard list', () => {
  const action = dashboardsActions.setDashboardListOrder({ order: 'az' });
  const { dashboardListOrder } = dashboardsReducer(initialState, action);

  expect(dashboardListOrder).toBe('az');
});

test('set public access to the dashboard', () => {
  const dashboardId = '@dashboard/01';
  const isPublicTest = false;
  const accessKey = 'public-access-key';

  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      data: dashboardsMeta,
    },
  };
  const action = dashboardsActions.setDashboardPublicAccess({
    dashboardId,
    isPublic: isPublicTest,
    accessKey,
  });
  const {
    metadata: { data },
  } = dashboardsReducer(state, action);

  const { isPublic, publicAccessKey } = data.find(
    (item) => item.id === dashboardId
  );

  expect(isPublic).toEqual(isPublicTest);
  expect(publicAccessKey).toEqual(accessKey);
});

test('regenerate access key for the dashboard', () => {
  const dashboardId = '@dashboard/01';
  const action = dashboardsActions.regenerateAccessKey({ dashboardId });

  expect(action).toMatchInlineSnapshot(`
    Object {
      "payload": Object {
        "dashboardId": "@dashboard/01",
      },
      "type": "dashboards/regenerateAccessKey",
    }
    `);
});

test('add cloned dashboard to the list', () => {
  const newDashboard = {
    id: '@dashboard/01',
    title: null,
    widgets: 0,
    queries: 0,
    tags: [],
    lastModificationDate: 1606895352390,
    isPublic: false,
    publicAccessKey: null,
    savedQueries: [],
  };
  const action = dashboardsActions.addClonedDashboard(newDashboard);
  const {
    metadata: { data },
  } = dashboardsReducer(initialState, action);

  expect(data).toMatchInlineSnapshot(`
    Array [
      Object {
        "id": "@dashboard/01",
        "isPublic": false,
        "lastModificationDate": 1606895352390,
        "publicAccessKey": null,
        "queries": 0,
        "savedQueries": Array [],
        "tags": Array [],
        "title": null,
        "widgets": 0,
      },
    ]
  `);
});

test('prepare tagsPool', () => {
  const state = {
    ...initialState,
    metadata: {
      ...initialState.metadata,
      isLoaded: true,
      data: [
        {
          id: '@dashboard/01',
          widgets: 1,
          queries: 0,
          title: null,
          lastModificationDate: 0,
          tags: ['marketing'],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
        {
          id: '@dashboard/02',
          widgets: 1,
          queries: 0,
          title: null,
          lastModificationDate: 0,
          tags: ['sales'],
          isPublic: false,
          publicAccessKey: null,
          savedQueries: [],
        },
      ],
    },
  };

  const action = dashboardsActions.prepareTagsPool();
  const { tagsPool } = dashboardsReducer(state, action);

  expect(tagsPool).toMatchInlineSnapshot(`
    Array [
      "marketing",
      "sales",
    ]
  `);
});

test('clear tagsPool', () => {
  const action = dashboardsActions.clearTagsPool();
  const { tagsPool } = dashboardsReducer(initialState, action);

  expect(tagsPool).toMatchInlineSnapshot(`Array []`);
});

test('set tagsFilters tags', () => {
  const action = dashboardsActions.setTagsFilters({
    tags: ['sales', 'marketing'],
  });
  const {
    tagsFilters: { tags },
  } = dashboardsReducer(initialState, action);

  expect(tags).toMatchInlineSnapshot(`
    Array [
      "sales",
      "marketing",
    ]
  `);
});

test('set tagsFilters showOnlyPublicDashboards', () => {
  const action = dashboardsActions.setTagsFiltersPublic(true);
  const {
    tagsFilters: { showOnlyPublicDashboards },
  } = dashboardsReducer(initialState, action);

  expect(showOnlyPublicDashboards).toBeTruthy();
});

test('set dashboards connections', () => {
  const dashboards = [
    {
      id: '@id-1',
      title: '@dashboard-1',
    },
    {
      id: '@id-2',
      title: '@dashboard-2',
    },
    {
      id: '@id-3',
      title: '@dashboard-3',
    },
  ];
  const action = dashboardsActions.setConnectedDashboards(dashboards);
  const {
    connectedDashboards: { items },
  } = dashboardsReducer(initialState, action);

  expect(items).toEqual(dashboards);
});

test('set dashboards connections error', () => {
  const action = dashboardsActions.setConnectedDashboardsError(true);
  const {
    connectedDashboards: { isError },
  } = dashboardsReducer(initialState, action);

  expect(isError).toBeTruthy();
});

test('set dashboards connections loading', () => {
  const action = dashboardsActions.setConnectedDashboardsLoading(true);
  const {
    connectedDashboards: { isLoading },
  } = dashboardsReducer(initialState, action);

  expect(isLoading).toBeTruthy();
});
