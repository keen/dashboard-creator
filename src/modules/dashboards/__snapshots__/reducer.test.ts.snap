// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add widget to dashboard 1`] = `
Object {
  "cachedDashboardIds": Array [],
  "dashboardListOrder": "recent",
  "dashboardSettingsModal": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "dashboardShareModal": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "deleteConfirmation": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "items": Object {
    "@dashboard/01": Object {
      "error": null,
      "initialized": true,
      "isSaving": false,
      "settings": Object {
        "version": "0.0.1",
        "widgets": Array [
          "@widget/01",
        ],
      },
    },
  },
  "metadata": Object {
    "data": Array [
      Object {
        "id": "@dashboard/01",
        "isPublic": false,
        "lastModificationDate": 0,
        "publicAccessKey": null,
        "queries": 0,
        "tags": Array [],
        "title": null,
        "widgets": 1,
      },
    ],
    "error": null,
    "isInitiallyLoaded": false,
    "isLoaded": true,
    "isSavingMetadata": false,
  },
  "tagsFilters": Object {
    "showOnlyPublicDashboards": false,
    "tags": Array [],
  },
  "tagsPool": Array [],
}
`;

exports[`removes widget from dashboard 1`] = `
Object {
  "cachedDashboardIds": Array [],
  "dashboardListOrder": "recent",
  "dashboardSettingsModal": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "dashboardShareModal": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "deleteConfirmation": Object {
    "dashboardId": null,
    "isVisible": false,
  },
  "items": Object {
    "@dashboard/01": Object {
      "error": null,
      "initialized": true,
      "isSaving": false,
      "settings": Object {
        "version": "0.0.1",
        "widgets": Array [],
      },
    },
  },
  "metadata": Object {
    "data": Array [
      Object {
        "id": "@dashboard/01",
        "isPublic": false,
        "lastModificationDate": 0,
        "publicAccessKey": null,
        "queries": 0,
        "tags": Array [],
        "title": null,
        "widgets": 0,
      },
    ],
    "error": null,
    "isInitiallyLoaded": false,
    "isLoaded": true,
    "isSavingMetadata": false,
  },
  "tagsFilters": Object {
    "showOnlyPublicDashboards": false,
    "tags": Array [],
  },
  "tagsPool": Array [],
}
`;
